libleptongui/scheme/schematic/builtins.scm

```diff
@@ -428,7 +428,89 @@ the snap grid size should be set to 100")))
           (page_select_widget_update *window)))))
 
 
-(define-action-public (&edit-update #:label (G_ "Update Component") #:icon "gtk-refresh")
+;;; Update a component.
+(define-action-public (&edit-update #:label (G_ "Update Component")
+                                    #:icon "gtk-refresh")
+  ;; Update component to the latest version of the symbol
+  ;; available in the symbol library, while preserving any
+  ;; attributes set in the current schematic. On success, the
+  ;; component is deleted on its page and replaced with a new
+  ;; one. On failure, the current component is left unchanged.
+  (define (update-component *window current-component)
+    (let ((basename (component-basename current-component))
+          (page (object-page current-component))
+          (position (component-position current-component))
+          (angle (component-angle current-component))
+          (mirror? (component-mirror? current-component))
+          (locked? (component-locked? current-component))
+          (embedded? (object-embedded? current-component))
+          (attribs (object-attribs current-component)))
+
+      ;; Force symbol data to be reloaded from source
+      (s_clib_symbol_invalidate_data
+       (s_clib_get_symbol_by_name (string->pointer basename)))
+
+      ;; Create new object.
+      (let ((new-component
+             (set-component-with-transform!
+              (make-component/library basename
```

Comment:
`make-component/library()` could fail at this point, and
`set-component-with-transform!()` will throw an error:

```
Could not eval key "U": wrong-type-arg (#f "Wrong type argument in position ~A (expecting ~A): ~A" (1 component #f) #f)
```

It could happen if one opens another schematic from the directory
with `gafrc` calling `reset-component-library()` and setting up
different component library paths. Then there's a chance that components
user wants to update in previously opened schematic(s) are no longer available.

Reply:
I managed to reproduce the issue, thank you!
Hopefully the last commit solves the issue.

Reply:
Yes, thank you.